scripts/lib/releaseNotes.ts

```diff
@@ -150,3 +149,161 @@ function addNewReleaseNoteToc(releaseNotesNode: any, newVersion: string) {
+          return -1;
+        } else if (version1.length > version2.length) {
+          return 1;
+        }
+      }
+
+      return version2.localeCompare(version1);
+    },
+  );
+
+  return Object.fromEntries(markdownByPatchSorted);
+}
+
+/**
+ * Process the markdown dividing it into small sections for each patch version
+ *
+ * Returns a tuple:
+ *   1. A Set of strings representing all the minor versions we found in the
+ *      markdown.
+ *   2. A dictionary with the corresponding markdown for each patch version
+ *      found.
+ */
+export function extractMarkdownReleaseNotesPatches(
+  markdown: string,
+): [Set<string>, { [id: string]: string }] {
+  const sectionsSplit = markdown.split(/\n## (?=[0-9])/);
+  const sections: string[] = sectionsSplit.slice(1, sectionsSplit.length);
+
+  const minorVersionsFound = new Set<string>();
+  const markdownByPatchVersion: { [id: string]: string } = {};
+
+  for (let section of sections) {
+    const versionPatch = section.split("\n").slice(0, 1)[0];
+    const versionMinor = versionPatch.split(".").slice(0, 2).join(".");
+
+    minorVersionsFound.add(versionMinor);
+
+    const content = section.split("\n");
+    content.shift();
+    markdownByPatchVersion[versionPatch] = `## ${versionPatch}\n${content.join(
+      "\n",
+    )}`;
+  }
+
+  return [minorVersionsFound, markdownByPatchVersion];
+}
+
+/**
+ * Updates the release notes folder by adding the notes to their corresponding version
+ * file.
+ */
+export async function writeSeparateReleaseNotes(
+  pkg: Pkg,
+  releaseNoteMarkdown: string,
+): Promise<void> {
+  if (!pkg.hasSeparateReleaseNotes) {
+    throw new Error(
+      `The package ${pkg.name} doesn't have separate release notes`,
+    );
+  }
+
+  // Dictionary to store the file header in case we need to reconstruct a file from a
+  // previous version
+  const filesToInitialHeaders: { [id: string]: string } = {};
+  const basePath = `${getRoot()}/docs/api/${pkg.name}/release-notes`;
+
+  const [minorVersionsFound, markdownByPatchVersion] =
+    extractMarkdownReleaseNotesPatches(releaseNoteMarkdown);
+
+  // Read the current release notes for each version found
+  for (let version of minorVersionsFound) {
+    const versionPath = `${basePath}/${version}.md`;
+
+    if (!(await pathExists(versionPath))) {
+      // We don't have any release note file for that version
+      continue;
+    }
+
+    const currentMarkdown = await readFile(versionPath, "utf-8");
+    filesToInitialHeaders[version] = currentMarkdown
+      .split(/\n## (?=[0-9])/)
+      .slice(0, 1)[0];
+
+    const [_, markdownByPatchOldVersion] =
+      extractMarkdownReleaseNotesPatches(currentMarkdown);
+
+    for (let [versionPatch, markdownPatch] of Object.entries(
+      markdownByPatchOldVersion,
+    )) {
+      // We keep the release notes for a patch if it hasn't been modified for the current release notes.
+      // Otherwise, we keep the modified version
```

Comment:
Nit: `keep` suggests to me that we are not modifying anything

```suggestion
      // Otherwise, we use the modified version.
```